File: frontend/extract_script.py
import re
import subprocess

html_path = r"c:\Users\mizin\2ndBrain\SecondBrain\frontend\second-brain.html"
js_out_path = r"c:\Users\mizin\2ndBrain\SecondBrain\frontend\test_script.js"

with open(html_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Find the main script tag starting from line 1092 or matching <script> after details
script_matches = list(re.finditer(r'<script>', content))
# The last <script> tag should be the main one
if script_matches:
    last_script_start = script_matches[-1].end()
    # Find the corresponding closing </script> tag
    closing_match = content.find('</script>', last_script_start)
    if closing_match != -1:
        script_code = content[last_script_start:closing_match]
        with open(js_out_path, 'w', encoding='utf-8') as js_f:
            js_f.write(script_code)
        print(f"Successfully wrote main script section ({len(script_code)} bytes) to test_script.js")
    else:
        print("Could not find closing </script> tag")
else:
    print("Could not find any <script> tag")

File: extract_script.js
const fs = require('fs');
const path = require('path');

const htmlPath = path.join(__dirname, 'frontend', 'second-brain.html');
const jsOutPath = path.join(__dirname, 'frontend', 'test_script.js');

try {
  const content = fs.readFileSync(htmlPath, 'utf8');
  
  // Find all <script> start tags
  const regex = /<script>/g;
  let matches = [];
  let match;
  while ((match = regex.exec(content)) !== null) {
    matches.push(match.index);
  }
  
  if (matches.length > 0) {
    const lastScriptStart = matches[matches.length - 1] + '<script>'.length;
    const closingIndex = content.indexOf('</script>', lastScriptStart);
    
    if (closingIndex !== -1) {
      const scriptCode = content.substring(lastScriptStart, closingIndex);
      fs.writeFileSync(jsOutPath, scriptCode, 'utf8');
      console.log(`Successfully wrote main script section (${scriptCode.length} bytes) to test_script.js`);
      
      // Now let's spawn "node -c" to check syntax
      const { execSync } = require('child_process');
      try {
        execSync(`node -c "${jsOutPath}"`);
        console.log("✅ Syntax validation passed! No javascript syntax errors found.");
      } catch (syntaxError) {
        console.error("❌ Syntax validation failed! Error:");
        console.error(syntaxError.stderr ? syntaxError.stderr.toString() : syntaxError.message);
      }
    } else {
      console.error("Could not find closing </script> tag");
    }
  } else {
    console.error("Could not find any <script> tag");
  }
} catch (e) {
  console.error("Error running extractor:", e);
}
